docs/guides/model_surgery.md

```diff
@@ -0,0 +1,140 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.13.8
+  kernelspec:
+    display_name: Python 3 (ipykernel)
+    language: python
+    name: python3
+---
+
+Model Surgery
+==============================
+
+Usually, Flax modules and optimizers track and update the params for you. But there may be some time when you want to do some model surgery and tweak the param tensors yourself. This guide shows you how to do the trick.
+
+
+## Setup
+
+```python
+!pip install --upgrade -q pip jax jaxlib flax
+```
+
+```python
+import functools
+
+import jax
+import jax.numpy as jnp
+from flax import traverse_util
+from flax import linen as nn
+from flax.core import freeze
+import jax
+import optax
+```
+
+Surgery with Flax Modules
+--------------------------------
+
+Let's create a small convolutional neural network model for our demo.
+
+As usual, you can run `CNN.init(...)['params']` to get the `params` to pass and modify it in every step of your training.
+
+
+```python
+class CNN(nn.Module):
+    @nn.compact
+    def __call__(self, x):
+      x = nn.Conv(features=32, kernel_size=(3, 3))(x)
+      x = nn.relu(x)
+      x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))
+      x = nn.Conv(features=64, kernel_size=(3, 3))(x)
+      x = nn.relu(x)
+      x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))
+      x = x.reshape((x.shape[0], -1))
+      x = nn.Dense(features=256)(x)
+      x = nn.relu(x)
+      x = nn.Dense(features=10)(x)
+      x = nn.log_softmax(x)
+      return x
+
+def get_initial_params(rng):
```

Comment:
Should we use the `key` variable name here as well?

Reply:
Done.